### DISCUSSION TITLE
ContentTemplateSelector not working in WASM?

### DISCUSSION THREAD

**Discussion by KarlHedlund** [unoplatform/uno#10524] [Q&A] [answered]
I get different behaviour with WinUI and WASM when it comes to the ContentTemplateSelector on a ContentDialog. I am creating a way to choose the contents of a ContentDialog by using a ContentTemplateSelector. When running the WinUI project, the selector is working, but when running it in Edge or Brave, the selector is never used (checked with debugger). Instead, the Content property is used.

Am I missing something or is this a bug? I feel like the need to be able to control the contents of a ContentDialog is so fundamental that there would be more articles about it should it be a bug. I am just starting out with Uno, coming from WPF, so there might be a way to do things that I'm not aware of yet. Having to set the XamlRoot is new for me, so not sure if this is the proper way to prevent the error of not setting it.

### Code details
I've just created a new Uno Platform project with WinUI and WASM, and made the following changes:
In my real project it is set up with a ViewModel as content, but for demonstration purposes I've just set the content as a simple TextBlock. 

Running in the WinUI project, the dialog shows "If this shows up, the selector is active". If running WASM, "The selector was not used" is shown in the dialog.

```xaml
<ContentDialog
    x:Class="ContentTemplateSelectorProblem.CustomDialog"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:ContentTemplateSelectorProblem"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d"
    Title="Content Template Selector problem"
    PrimaryButtonText="OK"
    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
    
    <ContentDialog.Content>
        <TextBlock Text="The selector was not used."/>
    </ContentDialog.Content>
    
    <ContentDialog.ContentTemplateSelector>
        <local:TemplateSelector>
            <local:TemplateSelector.ATemplate>
                <DataTemplate>
                    <TextBlock Text="If this shows up, the selector is active."/>
                </DataTemplate>
            </local:TemplateSelector.ATemplate>
        </local:TemplateSelector>
    </ContentDialog.ContentTemplateSelector>

</ContentDialog>
```

The data template selector:
```CSharp
public class TemplateSelector : DataTemplateSelector
{
    public DataTemplate ATemplate { get; set; }

    protected override DataTemplate SelectTemplateCore(object item) => SelectTemplateCore(item, null);

    protected override DataTemplate SelectTemplateCore(object item, DependencyObject container)
    {
        return ATemplate;
    }
}
```

The code that opens the dialog:
```CSharp
private async void Page_Loaded(object sender, RoutedEventArgs e)
{
    var contentDialog = new CustomDialog() { XamlRoot = this.XamlRoot};
    var result = await contentDialog.ShowAsync();
}
```

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the report! Can you provide a self-contained repro with the code you posted here? It will make it easier to ensure that we're talking about the same issue.

**Reply by KarlHedlund** [reply]
> Thanks for the report! Can you provide a self-contained repro with the code you posted here? It will make it easier to ensure that we're talking about the same issue.

Sure, here you go: https://github.com/KarlHedlund/ContentTemplateSelectorProblem

**Reply by jeromelaban** [reply]
Thanks, let's track in https://github.com/unoplatform/uno/issues/10537, I'll post a workaround.
